geode-core/src/distributedTest/java/org/apache/geode/internal/cache/control/RebalanceOperationDistributedTest.java

```diff
@@ -2820,68 +2823,58 @@ private void doRebalance() throws TimeoutException, InterruptedException {
 
   private void doConcurrentPutInTx(String s) throws Exception {
     for (int i = 0; i < totalNumberOfBuckets; i++) {
-      synchronized (list) {
-        list.add(i);
-      }
-    }
-
-    Thread[] threads = new Thread[totalNumberOfBuckets];
-    for (int i = 0; i < totalNumberOfBuckets; i++) {
-      threads[i] = new Thread(() -> doPutOpInTx(s));
+      queue.add(i);
     }
 
+    ExecutorService pool = Executors.newCachedThreadPool();
+    Future<?>[] futures = new Future<?>[totalNumberOfBuckets];
     for (int i = 0; i < totalNumberOfBuckets; i++) {
-      threads[i].start();
+      futures[i] = pool.submit(() -> doPutOpInTx(s));
```

Comment:
I was more thinking in the line of using the `invokeAll` rather than the single `submit`. With `invokeAll` you can create all the operations up front and then invoke all of them at the same time.

Are we not concerned that if we do it this way, that there is a possibility that the concurrency issue is possibly not ever hit?